libraries/libgambc/0002-Precompiled-C-files-for-backport-WHEN-and-UNLESS.patch

```diff
@@ -0,0 +1,10632 @@
+From f6ece71a0ada833c2cd69baf7b1d17eeea2e212a Mon Sep 17 00:00:00 2001
```

Comment:
Same problem as the other patch. Also I don't know how to sanely review a 10000 line patch as this one is not just changing a few linker line numbers. Can we reproduce this by doing the other patch and then bootstrapping it?

Reply:
Neither do I know how to review a huge amount of generated code.

The generated `C` files are a result of these steps taken to produce
both, the source backport and the compiled files:

1. unpack gambc archive
2. `cd gambc-v4_7_9; git init; git add .; git commit -m 'initial import'`
3. `./configure; make`
4. edit `lib/_nonstd.scm` and backport the code from gambit 0.9.2
5. commit source
6. use `git format-patch` to create the first patch
7. `make from-scratch`
   This step compiles the
8. write a little test case in `testwhen.scm`:

    (display "Test when: ")
    (when (= 0 1) (display "comparsion ") (display "failed\n"))
    (unless (= 0 1) (display "comparsion ") (display "works\n"))
9. `/gsc/gsc   -:~~bin=./bin,~~lib=./lib,~~include=./include -exe testwhen.scm`
10. run `./testwhen` and verify result as `Test when: comparsion works`
11. run `./bin/gsi-script testwhen.scm`and find the same result
12. `git add` the modified `C` files
13. use `git format-patch` to create the precompiled patch
14. copy both patchs to `libgambc` directory
15. modify `make.sh`
16. check changes in and create PR